lib/utils/clipboard_utils.dart

```diff
@@ -86,4 +87,44 @@ class ClipboardUtils {
       }
     }
   }
+
+  /// Reads text from clipboard and shows appropriate toast messages
+  ///
+  /// [ref] - WidgetRef for accessing providers
+  /// [onPaste] - Callback function that receives the pasted text
+  /// [successMessage] - Optional custom message to show (defaults to "Pasted from clipboard")
+  /// [noTextMessage] - Optional custom message to show when clipboard is empty (defaults to "Nothing to paste from clipboard")
+  /// [errorMessage] - Optional custom error message to show when clipboard operation fails (defaults to "Clipboard unavailable")
+  static Future<void> pasteWithToast({
```

Comment:
This clipboard utils files had tests for other methods. I left a test proposal in draft, in case you like the idea fee free to cherry-pick that commit  #643 . If you don't is fine too 😉